### ISSUE TITLE
Uplift Header to include a Subheader option

### ISSUE THREAD

**Issue by claytonbeese** [cerner/terra-clinical#142]
# Issue Description
Adding a flag onto the Header control to allow it to be a 'subheader'.  Flag would change the background to grey-25, and possibly shrink the vertical padding of the header to differentiate it.  Needed for the Radiology worklist.

## Issue Type
- [ ] New Feature
- [x] Enhancement
- [ ] Bug
- [ ] Other

## Expected Behavior
Should behave exactly as a header.  Visual distinction will be used only to indicate a secondary level of hierarchy.

## Current Behavior
Header is a static background color and padding height, leading to a single level of hierarchy.

**Comment by claytonbeese** [comment]
![image](https://user-images.githubusercontent.com/28354462/28188076-774c2896-67e6-11e7-8255-044fd06ed801.png)

attaching an initial mockup of the subheader

**Comment by nagrawal3** [comment]
# Tech Design for terra-clinical-header
## Summary:
Terra-clinical-header will accept another flag isSubsection to make the header as sub-sectional header. To distinguish sub-sectional header with the header, background color of the sub-sectional header will be @grey25 i.e. #c8cacb and it will not have any border.

To keep this as passive change, isSubsection prop will have a default value as false.

## React Props:
| Prop     | Type | Default | Description  |
|----------|------|---------|--------------|
| startContent | element | null    | Content to be displayed at the start of the header, placed before the title |
| title | string | ''    | Text to be displayed as the title in the header bar |
| endContent | element | null    | Content to be displayed at the end of the header |
| isSubsection | Bool | false    | Should header be a sub-sectional header |

## Example:
```
import React from 'react';
import Button from 'terra-button';
import Header from 'terra-clinical-header';

const SubSectionalHeaderExampleComponent = () => (
  <div>
    <Header
      startContent={<div><Button text="Button" style={{ margin: '0 10px 0 0' }} /></div>}
      title="Default Header"
      endContent={<div><Button text="Button" style={{ margin: '0 0 0 10px' }} /></div>}
      isSubsection
    />
  </div>
);

export default SubSectionalHeaderExampleComponent;
```

**Comment by emilyrohrbough** [comment]
Nit: can we use the name `isSubheader` instead of `isSubction`

**Comment by nagrawal3** [comment]
I can use `isSubheader` as the prop.

AS per UX design, font size will be same as we have for header.

**Comment by tbiethman** [comment]
@claytonbeese would you expect the start/endContent props to be usable with a subheader? It seems like that might get a little weird due to the subheader's lesser height.

**Comment by claytonbeese** [comment]
Right now, I wasn't expecting it to shrink (though that might be handy in the future).  What I remember of the legacy terra subheader was that it was the smaller height until a button was added, at which point it expanded to comfortably hold that button.

So yes, I would expect it to use the start/endContent props.

**Comment by neilpfeiffer** [comment]
@claytonbeese and I talked through the initial details, for this enhancement, lets keep the full capability of current header available for subheader, invoked by adding the new prop, correctly using an appropriate h1-h6 based on order/index for screen-readers, and with the only style changes for now as:

* background-color: $terra-grey-25 (#C8CACB)
* border-bottom-width: 0;
* border-bottom-style: none;
* border-bottom-color: initial;
* padding: 0.3571rem 0.7143rem;
* height: 2.857rem;

We'll consider this a minor enhancement and if we need to adjust additional styling (font-weight, font-size, etc.), we can handle that in a future issue. 

Since I did not ever create an initial Terra UX design file for Clinical-Header (Jordan will be helping me work backfill the missing design files) - lets consider the information in this issue the approved UX Input to get this enhancement started. I'm sure the team is eager to start contributing the work, since the uplift to CSS modules is now complete.
